### PR TITLE
Add explanations about features to improve UX

### PR DESCRIPTION
**Pull request by glena**
Add copy to explain what the different features are as the user configure deployment settings.

If the APP is not installed and the repo url points to github

image

If the APP is installed and the repo url points to github

image

When asking for OIDC configuration

image

It also adds a hint to accept multiselect prompts by hitting enter

image

Fix https://github.com/pulumi/pulumi-service/issues/21604 
Fix https://github.com/pulumi/pulumi-service/issues/21606 
Fix https://github.com/pulumi/pulumi-service/issues/21609 
Fix https://github.com/pulumi/pulumi-service/issues/21610 
Fix https://github.com/pulumi/pulumi-service/issues/21611

### CODE REVIEW

**Comment by cleverguy25** [comment]
For the first one, rather than forcing them to abort, should we ask them if they want to abort?

**Comment by glena** [comment]
@cleverguy25 done, let me know if I need to adjust the copy https://github.com/pulumi/pulumi/pull/17307/commits/2a40f66f208e10c8354ed82dc567a510f9098071

File: pkg/cmd/pulumi/deployment_settings_config.go [hunk 1]
```diff
@@ -251,6 +260,18 @@ func initStackDeploymentCmd(
 		return err
 	}
 
+	oidcExplanationMsg := `
+Pulumi supports OpenID Connect (OIDC) integration across various services by
+leveraging signed, short-lived tokens and eliminating the necessity for hardcoded
+cloud provider credentials and facilitates the exchange of these tokens for
+short-term credentials.
```
Comment (julienp) [inline pkg/cmd/pulumi/deployment_settings_config.go:267]: Should we keep this and the other messages on a single line and let the terminal do the wrapping?

File: pkg/cmd/pulumi/deployment_settings_config.go [hunk 2]
```diff
@@ -432,8 +456,58 @@ func configureGit(d *deploymentSettingsCommandDependencies, gitSSHPrivateKeyPath
...
+`
+
+		ghAppExplanationMsg = colors.Highlight(ghAppExplanationMsg, "Pulumi’s GitHub app", colors.SpecHeadline)
+		ghAppExplanationMsg = colors.Highlight(ghAppExplanationMsg, "Pulumi Deployments", colors.SpecHeadline)
+
+		d.Prompts.Print(d.DisplayOptions.Color.Colorize(ghAppExplanationMsg))
+
 		useGitHub = d.Prompts.AskForConfirmation(
-			"A GitHub repository was detected, do you want to use the Pulumi GitHub App?",
+			"do you want to use the Pulumi GitHub App?",
```
Comment (julienp) [inline pkg/cmd/pulumi/deployment_settings_config.go:510]:
```suggestion
			"Do you want to use the Pulumi GitHub App?",
```

File: pkg/cmd/pulumi/deployment_settings_config.go (hunk 1 above)
Comment (julienp) [inline pkg/cmd/pulumi/deployment_settings_config.go:267]: Should we leave this message on a single line and let the terminal to the wrapping? Similarly for the other messages below.

Comment (glena) [reply pkg/cmd/pulumi/deployment_settings_config.go:267]: sure, I thought for such long messages it was easier to read incases the terminal is to wide (and in some cases some messages are formated this way here). Will change

Comment (glena) [reply pkg/cmd/pulumi/deployment_settings_config.go]: done

**Review by julienp — APPROVED** [review]

**Pull request merged**